sys/kern/kern_cheri_revoke.c

```diff
@@ -650,6 +664,8 @@ kern_cheri_revoke(struct thread *td, int flags,
 	/* Per-process kernel hoarders */
 	cheri_revoke_hoarders(td->td_proc, &vmcrc);
 
+	cheri_revoke_map_hoards(td->td_proc, &vmcrc);
```

Comment:
What happens if a process has more shared memory associated with it than fits in its address space (assuming that's possible)?

Reply:
In my initial commit push I called panic. I'm now killing the process since revocation can't succeed. In practice you'd have to be doing something fairly ridiculous to get into this state so I'm not all that worried about being rude to the process.